### ISSUE TITLE
Automatic external documents retreiving

### ISSUE THREAD

**Issue by CkuT** [the-paperless-project/paperless#598]
Hey,

I have a small personal [scrapy](https://scrapy.org) side project whose purpose is to retrieve each month any kind of my documents (ISP bills, energy supplier bills, payslips, ...), with some extra features (deduplication, automatically renaming, ...). I built this project for my paperless instance, so each of this file is properly named to be parsed by paperless, which automates a lot of work for me each month.

I wonder if anyone would be interested to have something like that as a paperless feature ? (maybe as a paperless plugin/django app ?). I know some people are keeping their paperless instance outside of any internet access, so I'd like to gather some opinions on this before investing some time :)

**Comment by bauerj** [comment]
This does sound interesting. Can you provide more details? Is it necessary to write code for each source to retrieve the documents?

**Comment by CkuT** [comment]
Yep, each source is specific and a spider must be written for it. The complex part is the login phase: once it is done, the information is often available on a rest API or on a simple HTML page.

Fortunately, I did not have to handle complex logics (2FA, JavaScript interpretation, ...), but some services may need to implement these kind of sequences.
